syncer/malsync/syncer.go

```diff
@@ -341,8 +340,15 @@ func (s *Syncer) downloadNodeIDs(ctx context.Context, initial bool, updates chan
 	}
 }
 
-func (s *Syncer) updateState() error {
-	if err := malsync.UpdateSyncState(s.localdb, s.clock.Now()); err != nil {
+func (s *Syncer) updateState(ctx context.Context) error {
+	if err := s.localdb.WithTx(ctx, func(tx sql.Transaction) error {
+		return malsync.UpdateSyncState(tx, s.clock.Now())
+	}); err != nil {
+		if ctx.Err() != nil {
+			// FIXME: with crawshaw, canceling the context which has been used to get
+			// a connection from the pool may cause "database: no free connection" errors
```

Comment:
Is this tracked as an issue? I only took a quick look at this issue and I'm not sure I fully understand the issue, or rather I believe we are not using `crawshaw` correctly in that aspect.

Reply:
Adding a mention of #6273 to the comment in the scope of which we need to check how well context cancellation is handled by crawshaw